### DISCUSSION TITLE
Is there a way to access global coordinates?

### DISCUSSION THREAD

**Discussion by leo2r** [pyvista/pyvista#3173] [Q&A] [answered]
What coordinate system are meshes stored as in PyVista? I'm using it to create a Gym env for a reinforcement learning agent and I would like to keep track of the origin and distance from the origin (global coordinates) for resetting meshes to their original position, or clipping the agent's actions (which move the meshes around). Is this currently possible in PyVista? Or VTK? Or do I need to do it manually by keeping a running tab of each action translating the mesh? Thanks!

**Comment by tkoyama010** [comment, marked as answer]
When using the DataSet class, the location of each point in the mesh can be obtained with the [points](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.points.html) method.

**Reply by leo2r** [reply]
I'm looking more for the coordinates of the centre or origin of the mesh in relation to the global axis, does such a variable exist? Thank you!

**Reply by tkoyama010** [reply]
You can get the center coordinates of the mesh with the [center ](https://dev.pyvista.org/api/core/_autosummary/pyvista.DataSet.center.html?highlight=center#pyvista.DataSet.center) property.
